<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Retrieve gene orthologs/homologs — orthologs • babelgene</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Retrieve gene orthologs/homologs — orthologs"><meta property="og:description" content="Retrieve gene orthologs/homologs for a set of genes.
Converts between human and non-human analogs."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-130263433-2"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-130263433-2');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">babelgene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">22.3.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"></ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="../articles/babelgene-intro.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
<li>
  <a href="https://github.com/igordot/babelgene/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Retrieve gene orthologs/homologs</h1>
    <small class="dont-index">Source: <a href="https://github.com/igordot/babelgene/blob/HEAD/R/functions.R" class="external-link"><code>R/functions.R</code></a></small>
    <div class="hidden name"><code>orthologs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Retrieve gene orthologs/homologs for a set of genes.
Converts between human and non-human analogs.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">orthologs</span><span class="op">(</span><span class="va">genes</span>, <span class="va">species</span>, human <span class="op">=</span> <span class="cn">TRUE</span>, min_support <span class="op">=</span> <span class="fl">3</span>, top <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>genes</dt>
<dd><p>A vector of gene symbols or Entrez/Ensembl IDs.</p></dd>


<dt>species</dt>
<dd><p>Species name, such as <code>Mus musculus</code> or <code>mouse</code> (see <code><a href="species.html">species()</a></code> for options).</p></dd>


<dt>human</dt>
<dd><p>A logical scalar indicating if the input genes are human. If <code>TRUE</code>, the input genes are human. If <code>FALSE</code>, the input genes correspond to the non-human species and the output will be the human equivalents.</p></dd>


<dt>min_support</dt>
<dd><p>Minimum number of supporting source databases.</p></dd>


<dt>top</dt>
<dd><p>For each gene, output only the match with the highest support level if there are multiple hits.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A data frame of gene pairs (human and given species).</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Wright MW, Eyre TA, Lush MJ, Povey S, Bruford EA. HCOP: the HGNC comparison of orthology predictions search tool. <em>Mamm Genome</em>. 2005 Nov;16(11):827-8. <a href="https://doi.org/10.1007/s00335-005-0103-2" class="external-link">doi:10.1007/s00335-005-0103-2</a></p>
<p>Eyre TA, Wright MW, Lush MJ, Bruford EA. HCOP: a searchable database of human orthology predictions. <em>Brief Bioinform</em>. 2007 Jan;8(1):2-5. <a href="https://doi.org/10.1093/bib/bbl030" class="external-link">doi:10.1093/bib/bbl030</a></p>
<p>Seal RL, Gordon SM, Lush MJ, Wright MW, Bruford EA. genenames.org: the HGNC resources in 2011. <em>Nucleic Acids Res</em>. 2011 Jan;39:D514-9. <a href="https://doi.org/10.1093/nar/gkq892" class="external-link">doi:10.1093/nar/gkq892</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="fu">orthologs</span><span class="op">(</span>genes <span class="op">=</span> <span class="st">"TP53"</span>, species <span class="op">=</span> <span class="st">"mouse"</span>, human <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   human_symbol human_entrez   human_ensembl taxon_id symbol entrez</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1         TP53         7157 ENSG00000141510    10090  Trp53  22059</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              ensembl</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 ENSMUSG00000059552</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                         support</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   support_n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1        12</span>
<span class="r-in"><span class="fu">orthologs</span><span class="op">(</span>genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Ptprc"</span>, <span class="st">"Cd34"</span><span class="op">)</span>, species <span class="op">=</span> <span class="st">"mouse"</span>, human <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   human_symbol human_entrez   human_ensembl taxon_id symbol entrez</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1         CD34          947 ENSG00000174059    10090   Cd34  12490</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2        PTPRC         5788 ENSG00000081237    10090  Ptprc  19264</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              ensembl</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 ENSMUSG00000016494</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 ENSMUSG00000026395</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                         support</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                   EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   support_n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1        12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2        10</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Igor Dolgalev.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.</p>
</div>

      </footer></div>

  


  

  </body></html>

