<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to babelgene â€¢ babelgene</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to babelgene">
<meta property="og:description" content="babelgene">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-130263433-2"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-130263433-2');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">babelgene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">21.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"></ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="../articles/babelgene-intro.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
<li>
  <a href="https://github.com/igordot/babelgene/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="babelgene-intro_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to babelgene</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/igordot/babelgene/blob/master/vignettes/babelgene-intro.Rmd"><code>vignettes/babelgene-intro.Rmd</code></a></small>
      <div class="hidden name"><code>babelgene-intro.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>Genomic analysis of model organisms often requires the use of databases based on human data or making comparisons to patient-derived resources. This requires converting genes between human and non-human equivalents. The <code>babelgene</code> R package helps to simplify the process. It provides provides gene orthologs/homologs:</p>
<ul>
<li>in an R-friendly tidy/long format with one gene pair per row</li>
<li>for multiple frequently studied model organisms, such as mouse, rat, fly, and zebrafish</li>
<li>sourced from multiple databases</li>
<li>as gene symbols, NCBI Entrez, and Ensembl IDs</li>
<li>without a need for an active internet connection</li>
</ul>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<p>You can install the <code>babelgene</code> R package from CRAN.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"babelgene"</span><span class="op">)</span></code></pre></div>
<p>Load <code>babelgene</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/babelgene/">babelgene</a></span><span class="op">)</span></code></pre></div>
<p>The main functionality is accessed via the <code><a href="../reference/orthologs.html">orthologs()</a></code> function which takes one or more genes and outputs a data frame of predicted ortholog/homolog pairs. The output data frame contains gene symbols and IDs for human and the specified species. Several examples are provided below.</p>
<p>Get mouse equivalents for a set of human genes.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/orthologs.html">orthologs</a></span><span class="op">(</span>genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TP53"</span>, <span class="st">"EGFR"</span>, <span class="st">"IL6"</span>, <span class="st">"TGFB1"</span>, <span class="st">"CD4"</span><span class="op">)</span>, species <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span>
<span class="co">#&gt;   human_symbol human_entrez   human_ensembl taxon_id symbol entrez</span>
<span class="co">#&gt; 1          CD4          920 ENSG00000010610    10090    Cd4  12504</span>
<span class="co">#&gt; 2         EGFR         1956 ENSG00000146648    10090   Egfr  13649</span>
<span class="co">#&gt; 3          IL6         3569 ENSG00000136244    10090    Il6  16193</span>
<span class="co">#&gt; 4        TGFB1         7040 ENSG00000105329    10090  Tgfb1  21803</span>
<span class="co">#&gt; 5         TP53         7157 ENSG00000141510    10090  Trp53  22059</span>
<span class="co">#&gt;              ensembl</span>
<span class="co">#&gt; 1 ENSMUSG00000023274</span>
<span class="co">#&gt; 2 ENSMUSG00000020122</span>
<span class="co">#&gt; 3 ENSMUSG00000025746</span>
<span class="co">#&gt; 4 ENSMUSG00000002603</span>
<span class="co">#&gt; 5 ENSMUSG00000059552</span>
<span class="co">#&gt;                                                                                         support</span>
<span class="co">#&gt; 1 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span>
<span class="co">#&gt; 2 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span>
<span class="co">#&gt; 3                Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoMCL|Panther|PhylomeDB|Treefam</span>
<span class="co">#&gt; 4 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span>
<span class="co">#&gt; 5 EggNOG|Ensembl|HGNC|HomoloGene|Inparanoid|NCBI|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span>
<span class="co">#&gt;   support_n</span>
<span class="co">#&gt; 1        12</span>
<span class="co">#&gt; 2        12</span>
<span class="co">#&gt; 3        10</span>
<span class="co">#&gt; 4        12</span>
<span class="co">#&gt; 5        12</span></code></pre></div>
<p>Input genes are assumed to be human by default. You can specify if the input genes are human with the <code>human</code> parameter.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/orthologs.html">orthologs</a></span><span class="op">(</span>genes <span class="op">=</span> <span class="st">"Pu"</span>, species <span class="op">=</span> <span class="st">"fruit fly"</span>, human <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt;   human_symbol human_entrez   human_ensembl taxon_id symbol entrez     ensembl</span>
<span class="co">#&gt; 1         GCH1         2643 ENSG00000131979     7227     Pu  37415 FBgn0003162</span>
<span class="co">#&gt;                                                                               support</span>
<span class="co">#&gt; 1 EggNOG|Ensembl|HomoloGene|Inparanoid|OMA|OrthoDB|OrthoMCL|Panther|PhylomeDB|Treefam</span>
<span class="co">#&gt;   support_n</span>
<span class="co">#&gt; 1        10</span></code></pre></div>
<p>It is possible to search by NCBI Entrez or Ensembl IDs instead of gene symbols.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/orthologs.html">orthologs</a></span><span class="op">(</span>genes <span class="op">=</span> <span class="st">"ENSG00000111640"</span>, species <span class="op">=</span> <span class="st">"mouse"</span>, human <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt;   human_symbol human_entrez   human_ensembl taxon_id symbol entrez</span>
<span class="co">#&gt; 1        GAPDH         2597 ENSG00000111640    10090  Gapdh  14433</span>
<span class="co">#&gt;              ensembl                                                   support</span>
<span class="co">#&gt; 1 ENSMUSG00000057666 HGNC|HomoloGene|NCBI|OMA|OrthoDB|OrthoMCL|Panther|Treefam</span>
<span class="co">#&gt;   support_n</span>
<span class="co">#&gt; 1         8</span></code></pre></div>
<p>The <code><a href="../reference/orthologs.html">orthologs()</a></code> function requires the <code>species</code> parameter to be set. You can check all the species that can be queried with the help of the <code><a href="../reference/species.html">species()</a></code> function.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/species.html">species</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;    taxon_id                 scientific_name</span>
<span class="co">#&gt; 1     28377             Anolis carolinensis</span>
<span class="co">#&gt; 2      9913                      Bos taurus</span>
<span class="co">#&gt; 3      6239          Caenorhabditis elegans</span>
<span class="co">#&gt; 4      9615          Canis lupus familiaris</span>
<span class="co">#&gt; 5      7955                     Danio rerio</span>
<span class="co">#&gt; 6      7227         Drosophila melanogaster</span>
<span class="co">#&gt; 7      9796                  Equus caballus</span>
<span class="co">#&gt; 8      9685                     Felis catus</span>
<span class="co">#&gt; 9      9031                   Gallus gallus</span>
<span class="co">#&gt; 10     9544                  Macaca mulatta</span>
<span class="co">#&gt; 11    13616           Monodelphis domestica</span>
<span class="co">#&gt; 12    10090                    Mus musculus</span>
<span class="co">#&gt; 13     9258        Ornithorhynchus anatinus</span>
<span class="co">#&gt; 14     9598                 Pan troglodytes</span>
<span class="co">#&gt; 15    10116               Rattus norvegicus</span>
<span class="co">#&gt; 16     4932        Saccharomyces cerevisiae</span>
<span class="co">#&gt; 17   284812 Schizosaccharomyces pombe 972h-</span>
<span class="co">#&gt; 18     9823                      Sus scrofa</span>
<span class="co">#&gt; 19     8364              Xenopus tropicalis</span>
<span class="co">#&gt;                                                       common_name</span>
<span class="co">#&gt; 1                                     Carolina anole, green anole</span>
<span class="co">#&gt; 2   bovine, cattle, cow, dairy cow, domestic cattle, domestic cow</span>
<span class="co">#&gt; 3                                                       roundworm</span>
<span class="co">#&gt; 4                                                       dog, dogs</span>
<span class="co">#&gt; 5               leopard danio, zebra danio, zebra fish, zebrafish</span>
<span class="co">#&gt; 6                                                       fruit fly</span>
<span class="co">#&gt; 7                                   domestic horse, equine, horse</span>
<span class="co">#&gt; 8                                         cat, cats, domestic cat</span>
<span class="co">#&gt; 9                    bantam, chicken, chickens, Gallus domesticus</span>
<span class="co">#&gt; 10 rhesus macaque, rhesus macaques, Rhesus monkey, rhesus monkeys</span>
<span class="co">#&gt; 11                                      gray short-tailed opossum</span>
<span class="co">#&gt; 12                                             house mouse, mouse</span>
<span class="co">#&gt; 13              duck-billed platypus, duckbill platypus, platypus</span>
<span class="co">#&gt; 14                                                     chimpanzee</span>
<span class="co">#&gt; 15                               brown rat, Norway rat, rat, rats</span>
<span class="co">#&gt; 16                   baker's yeast, brewer's yeast, S. cerevisiae</span>
<span class="co">#&gt; 17                                                           &lt;NA&gt;</span>
<span class="co">#&gt; 18                                    pig, pigs, swine, wild boar</span>
<span class="co">#&gt; 19                      tropical clawed frog, western clawed frog</span></code></pre></div>
</div>
<div id="details" class="section level2">
<h2 class="hasAnchor">
<a href="#details" class="anchor"></a>Details</h2>
<p>The package is based on the data provided by the Human Genome Organization (HUGO) Gene Nomenclature Committee (HGNC) at the European Bioinformatics Institute. The HGNC Comparison of Orthology Predictions (HCOP) integrates the orthology assertions predicted for human genes by eggNOG, Ensembl Compara, HGNC, HomoloGene, Inparanoid, NCBI Gene Orthology, OMA, OrthoDB, OrthoMCL, Panther, PhylomeDB, TreeFam and ZFIN.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Igor Dolgalev.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
